time.js

```diff
@@ -1,12 +1,169 @@
-
 require('@js-joda/timezone');
 const time = require('@js-joda/core');
 
 //openHAB uses a RFC DateTime string, js-joda defaults to the ISO version, this defaults RFC instead
 const rfcFormatter = time.DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS[xxxx][xxxxx]");
 const targetParse = time.ZonedDateTime.prototype.parse;
+
+/**
+ * When formatter is not supplied, uses an RFC DateTimeFormatter that can parse the format
+ * of java.time.ZonedDateTime.toString()
+ * @param {String} text 
+ * @param {time.DateTimeFormatter} formatter 
+ * @returns {time.ZonedDateTime}
+ * @throws {DateTimeParseException} when the text cannot be parsed
+ */
 time.ZonedDateTime.prototype.parse = function (text, formatter = rfcFormatter) {
     return targetParse(text, formatter);
 }
 
-module.exports = time;
+/**
+ * A passthrough function that can allow users to call `toZonedDateTime` on
+ * almost anything that can be converted to a time.ZonedDateTime without first
+ * checking to see if it's already a time.ZonedDateTime. 
+ * @returns {time.ZonedDateTime} this
+ */
+time.ZonedDateTime.prototype.toZonedDateTime = function () {
+    return this;
+}
+
+/**
+ * Change the date of the passed in dt to match the date. This preserves
+ * the hours and minutes even in cases where dt and date span a DST changeover.
+ * @param {time.ZonedDateTime} dt date time to move the date on
+ * @param {time.ZonedDateTime} date date time to move dt's date to
+ * @returns {time.ZonedDateTime} dt with date's date
+ */
+const changeDate = function (dt, date) {
+    return dt.withYear(date.year())
+             .withMonth(date.month())
+             .withDayOfMonth(date.dayOfMonth());
+}
+
+/**
+ * Moves the date portion of the date time to today, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with today's date
+ */
+time.ZonedDateTime.prototype.toToday = function () {
+    return changeDate(this, time.ZonedDateTime.now());
+}
+
+/**
+ * Moves the date portion of the date time to tomorrow, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with tomorrow's date
+ */
+time.ZonedDateTime.prototype.toTomorrow = function () {
+    return changeDate(this, time.ZonedDateTime.now().plusDays(1));
+}
+
+/**
+ * Moves the date portion of the date time to yesterday, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with yesterday's date
+ */
+time.ZonedDateTime.prototype.toYesterday = function () {
+    return changeDate(this, time.ZonedDateTime.now().minusDays(1));
+}
+
+/**
+ * Compares this ZDT to see if it falls between start and end times,
+ * accounting for times that span midnight. start and end can be any type
+ * that has a `toZonedDateTime()` method but only the time portion is used.
+ * @param {*} start the starting time, anything that has a toZonedDateTime method
```

Comment:
This is another example of the whole point for the library. As long as the thing passed in has a `toZonedDateTime()` function that returns a `time.ZonedDateTime` it can be used here. 

- 500 : converted to a time.ZonedDateTime.now() 500 milliseconds into the future via the Number `toZonedDateTime` monkey patch
- 'PT5m`: converted to a time.ZonedDateTime.now() plus five minutes into the future via the String `toZonedDateTime` monkey patch
- `13:45`: converted to a time.ZonedDateTime.now() with the time adjusted to `13:45:00` via the String `toZonedDateTime` monkey patch
- `1:45 pm`: same as above
- `items.('MyTimeQuantityType')`: the state of the Item converted to a time.ZonedDateTime.now() plus the number seconds represented by the state (same for Dimmer Items), via the Items.toZonedDateTime() function
- `items.('MyNumber`): time.ZonedDateTime.now() plus the number of milliseconds represented by the state of the Item
- `items.('MyString')`: see the String examples above
- `items('MyDateTime'): the state of the Item converted to a time.ZonedDateTime via the Item.toZonedDateTime() function

This lets me do stuff like:

    if(time.ZonedDateTime.now().betweenTimes('22:00:00', '05:00:00')) // is it currently between 10pm and 5am?
    if(time.ZonedDateTime.now().betweenTimes(items.getItem('Sunset'), '8:00 am')) // is it currently between sunset and 8am?
    if(items.getItem('StartTime').toZonedDateTime().betweenTimes(time.ZonedDateTime().now(), 'PT1h')) // is the state of my DT Item between now and an hour from now?